### ISSUE TITLE
Wrong content-type for code provenance data

### ISSUE THREAD

**Issue by sivachandran** [DataDog/dd-trace-py#9774]
### Summary of problem
Wrong content-type used for code provenance data in multipart form data. This causes web application firewall to consider the request as malformed and terminate the connection.

[This](https://github.com/DataDog/dd-trace-py/blob/2b14b3e4f5a700bbe7de119cd0846a1cb2b4ff52/ddtrace/profiling/exporter/http.py#L207) line confirms `application/json` being used for code provenance data.

DataDog's Ruby profiler is using `application/octet-stream` as content-type which is correct. Refer [this](https://github.com/DataDog/dd-trace-rb/issues/3767#issuecomment-2217325350) issue.

Also, the form data name of code provenance data is `code-provenance` whereas in Ruby profiler it is named as `code-provenance.json`. It is better to follow consistent name.

### Which version of dd-trace-py are you using?
2.9.1

### Which version of pip are you using?
24.0

### Which libraries and their versions are you using?

<details>
  <summary>`pip freeze`</summary>
attrs==23.2.0
bcrypt==4.1.2
blinker==1.7.0
bytecode==0.15.1
cattrs==23.2.3
click==8.1.7
ddsketch==3.0.1
ddtrace==2.9.1
Deprecated==1.2.14
envier==0.5.1
exceptiongroup==1.2.0
Flask==3.0.2
importlib-metadata==6.11.0
itsdangerous==2.1.2
Jinja2==3.1.3
jsonschema==4.21.1
jsonschema-specifications==2023.12.1
MarkupSafe==2.1.5
opentelemetry-api==1.22.0
packaging==23.2
protobuf==4.25.2
python-dotenv==1.0.1
referencing==0.33.0
rpds-py==0.17.1
setuptools==70.2.0
six==1.16.0
tenacity==8.2.3
typing_extensions==4.9.0
Werkzeug==3.0.1
wheel==0.43.0
wrapt==1.16.0
xmltodict==0.13.0
zipp==3.17.0
</details>

### How can we reproduce your problem?
Profile a python app and inspect the data generated by the profiler.

### What is the result that you get?
```
...
--be325d252a94bf37ec01120db49905ad
Content-Disposition: form-data; name="code-provenance"; filename="code-provenance.json"
Content-Type: application/json
...
```
### What is the result that you expected?
`Content-Type` with value `application/octet-stream`.

**Comment by emmettbutler** [comment]
Thanks for the report, @sivachandran. This seems like a fairly straightforward fix to apply. We'll address it, but please feel free to open a pull request that fixes it if you get to it before we do.

**Comment by taegyunkim** [comment]
Fixed by #9837
